homeassistant/components/mqtt/cover.py

```diff
@@ -235,26 +235,12 @@ async def _async_setup_entity(
 class MqttCover(MqttEntity, CoverEntity):
     """Representation of a cover that can be controlled using MQTT."""
 
+    _attr_is_closed: bool | None = None
+    _attributes_extra_blocked: frozenset[str] = MQTT_COVER_ATTRIBUTES_BLOCKED
     _default_name = DEFAULT_NAME
     _entity_id_format: str = cover.ENTITY_ID_FORMAT
-    _attributes_extra_blocked: frozenset[str] = MQTT_COVER_ATTRIBUTES_BLOCKED
-
-    def __init__(
-        self,
-        hass: HomeAssistant,
-        config: ConfigType,
-        config_entry: ConfigEntry,
-        discovery_data: DiscoveryInfoType | None,
-    ) -> None:
-        """Initialize the cover."""
-        self._position: int | None = None
-        self._state: str | None = None
-
-        self._optimistic: bool | None = None
-        self._tilt_value: int | None = None
-        self._tilt_optimistic: bool | None = None
-
-        MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
+    _optimistic: bool
+    _tilt_optimistic: bool
```

Comment:
Not sure if these should be set to None by default and should be nullable, if not I am happy with this PR

Reply:
They are set as part of the `MqttEntity` constructor